### ISSUE TITLE
[ENHANCEMENT]Computation of error in metrics inconsistent

### ISSUE THREAD

**Issue by wlberendrecht** [pastas/pastas#518]
### Description of issue
In the metrics methods, the user may provide either residual series (res) or combined simulated series (sim) / observed series (obs).

If sim and obs are provided, the model error (`res`) is calculated as `(sim - obs)`. However, if res is provided, residuals are currently provided as ml.residuals() when called from modelstats.Statistics class. In ml.residuals(), residuals are calculated as `(obs - sim)`.

### Suggested fix
My suggestion is to explicitly use the variable `err` in the metrics methods:
```
if res is None:
    err = sim – obs
else:
    err = -1 * res
```
Then continue working with `err` instead of `res`. 

### Impact of inconsistency
If metrics use squared or absolute residuals, this inconsistency has no impact on the result. But it is suggested to fix it anyhow to improve consistency and readability.

**Comment by mbakker7** [comment]
An interesting discussion of the difference between 'residual' and 'error' may be found here: https://stats.stackexchange.com/questions/342466/are-residuals-predicted-minus-actual-or-actual-minus-predicted

Borrowing some words from that discussion: The term "residual" implies that it's what's left over after the model has been optimized, i.e. observed-model. (Prediction) "error" implies that it's how much the model deviates from the observation, i.e. model-observation. If we want to use 'errors' in the metrics method, I suggest we are explict about that in the doc string. 

Incidentally, I like a residual plot better than an error plot, as it pretty much takes the model and makes it a straight line. 

Not sure if we there is any metric where we don't use the squared error or the absolute error. Do we report the 'mean error'?

**Comment by raoulcollenteur** [comment]
The word 'error' is in various fit metrics (Mae, sse, ) , so to be consistent I will update the code and docs.

- use `err` in methods internally
- compute `err` from res
- [ ] document in doctoring res does not equal err.
- [ ] give user warning if none of obs, sim, or res is provided (while we're at it).
- [ ] update a couple of equations in the docstrings.

Thanks for reporting the Issue @wlberendrecht, PR with fix coming up :)
Raoul

**Comment by raoulcollenteur** [comment]
Fixed in PR #521, closing issue after merge.
